### DISCUSSION TITLE
Reduce garbage when `TextRenderer.RenderText` invoked.

### DISCUSSION THREAD

**Discussion by Cn-mjt44** [SixLabors/Fonts#423] [Ideas]
When I try to draw the game fps by using this lib, it cause too many GC when I try to invoke `TextRenderer.RenderText` in every frame.
![{9B8393C4-CBB5-47F0-B363-6DD6644C7DC3}](https://github.com/user-attachments/assets/afff71fa-b487-438c-808b-c0fdc26fccf7)
Drawing Logic:
```csharp

            if(fill == null && outline == null)
            {
                fill = Brushes.Solid(Color.White);
                outline = Pens.Solid(Color.Black, 2);
            }
            Texture texture = TextureToShow;
            Image<Rgba32> image = InnerImage;
            image.Mutate(i => i.Clear(Color.Transparent));
            image.Mutate(i => i.DrawText(TextOptions, text, fill, outline));
            image.Mutate(i => i.Flip(FlipMode.Vertical));
            texture.SetData(image,0);
            base.DrawInRect();
```

**Comment by JimBobSquarePants** [comment]
I’m not sure what you expect me to do here?

**Reply by Cn-mjt44** [reply]
Everytime when I Invok DrawText. It create too many garbage instance that need to destory by GC. I think resone of that are because it create too many managed object in there after I check the source code.
For example, the object of `List<T>` did not create with capacity, and that will cause list extend array for `log(2,2*AddCount)` times.

**Reply by JimBobSquarePants** [reply]
PR?

**Reply by Cn-mjt44** [reply]
> PR?

On working, not finish. And how to build the unit test evn?

**Reply by JimBobSquarePants** [reply]
> And how to build the unit test evn

Thanks! I'm not quite sure I understand the question. Are you asking how will write a unit test this? I think just a before and after demonstrating GC difference will suffice. The existing tests should cover breaking changes.

**Reply by Cn-mjt44** [reply]
> > And how to build the unit test evn
> 
> Thanks! I'm not quite sure I understand the question. Are you asking how will write a unit test this? I think just a before and after demonstrating GC difference will suffice. The existing tests should cover breaking changes.

Pull requests#424
